### PR TITLE
Add workflow to run next CI

### PR DESCRIPTION
**Pull request by FabioRosado**
## Description

This PR adds a workflow to run tests on the next branch

## Checklist

-   [ ] All tests pass locally
-   [ ] I have updated `docs/changelog.md`
-   [ ] I have created documentation for this(if applicable)

### CODE REVIEW

File: .github/workflows/test-next.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
+name: "[CI] Test Next"
+
+on:
+    push: # Only run on merges into main that modify files under pyscriptjs/ and examples/
+        branches:
+            - next
+        paths:
```
Comment (madhur-tandon) [inline .github/workflows/test-next.yml:7]: Are we sure these paths are correct?

Comment (FabioRosado) [reply .github/workflows/test-next.yml:7]: I copied the build workflow one so maybe?

Comment (madhur-tandon) [reply .github/workflows/test-next.yml:7]: hmm, the `next` branch has this new folder also, see https://github.com/pyscript/pyscript/tree/next
it's called `[pyscript.core](https://github.com/pyscript/pyscript/tree/next/pyscript.core)`

Looping in @WebReflection here...

Comment (WebReflection) [reply .github/workflows/test-next.yml:7]: core needs a story a part for integration and CI (currently integration is manual check) but `pyscriptjs` should always be tested as non breaking, imho, whenever any file changes in there ... does it make sense?

Comment (WebReflection) [reply .github/workflows/test-next.yml:7]: OK, pyscript core has a 100% code coverage so maybe that's also important moving forward, but integration (automated) is missing.

Comment (madhur-tandon) [reply .github/workflows/test-next.yml:7]: I think we are GTG @FabioRosado

**Review by madhur-tandon — APPROVED** [review]

**Pull request merged**